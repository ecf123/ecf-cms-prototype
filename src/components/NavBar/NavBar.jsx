import React, { useState } from 'react'
import Logo from '../Logo/Logo';
import NavBarItem from '../NavBarItem/NavBarItem';
import './NavBar.scss';

 const NavBar = () => {
  const [ selected, setSelected ] = useState("Dashboard");
  const getCurrentPage = (page) => {
    setSelected(page);
  }
  return (
    <div className='navBar'>
      <div className='navBar__container'>

        <div className='navBar__logo'>
          <Logo getCurrentPage={getCurrentPage} />
        </div>
        
        <ul className='navBar__linkList'>
          <li className='navBar__listItem'><NavBarItem imageSrc="" text="Dashboard" selected={selected} getCurrentPage={getCurrentPage} link="" /></li>
          <li className='navBar__listItem'><NavBarItem imageSrc="" text="Pathways" selected={selected} getCurrentPage={getCurrentPage} link="" /></li>
          <li className='navBar__listItem'><NavBarItem imageSrc="" text="Articles" selected={selected} getCurrentPage={getCurrentPage} link="" /></li>
          <li className='navBar__listItem'><NavBarItem imageSrc="" text="Marketplace" selected={selected} getCurrentPage={getCurrentPage} link="" /></li>
          <li className='navBar__listItem'><NavBarItem imageSrc="" text="Users" selected={selected} getCurrentPage={getCurrentPage} link="" /></li>
        </ul>

        <NavBarItem imageSrc="" text="Settings" selected={selected} getCurrentPage={getCurrentPage} link="" />
      </div>
    </div>
  )
}

export default NavBar;
