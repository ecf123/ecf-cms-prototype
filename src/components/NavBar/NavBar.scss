@use '../../styles/_variables' as *;
@use '../../styles/_typography' as *;
.navBar{

  &__container{
    display: grid;
    grid-template-rows: 1fr 4fr 2fr;
  }
}