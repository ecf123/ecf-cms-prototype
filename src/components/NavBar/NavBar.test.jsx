import { fireEvent, getByText } from "@testing-library/react";
import { customRender } from "../../utils/testUtils";
import NavBar from "./NavBar";

it("renders the NavBar", () => {
  const { container } = customRender(<NavBar />);
  expect(container).toMatchSnapshot();
});

it("shows Dashboard as selected on load", () => {
  const { container } = customRender(<NavBar />);

  const dashboardItem = getByText(container, "Dashboard").parentElement;

  expect(dashboardItem).toHaveStyle(
    "border-left: 3px solid $color-green-light;"
  );
});

it("shows Pathways as selected on when clicked", () => {
  const { container } = customRender(<NavBar />);

  const pathwaysItem = getByText(container, "Pathways");

  expect(pathwaysItem).toHaveStyle("color: $color-grey-dark;");

  fireEvent.click(pathwaysItem);

  expect(pathwaysItem).toHaveStyle("color: $color-blue-navy-dark;");
});
