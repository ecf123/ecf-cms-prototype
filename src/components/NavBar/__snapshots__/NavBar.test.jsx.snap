// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the NavBar 1`] = `
<div>
  <div
    class="nav-bar"
  >
    <div
      class="nav-bar__container"
    >
      <div
        class="nav-bar__logo"
      >
        <a
          class="link"
          href="/"
        >
          <div
            class="logo"
          >
            <div
              class="logo__box"
            >
              <img
                alt="The Future Hub Logo"
                class="logo__image"
                src="ECFP-company-logo.svg"
              />
            </div>
            <p
              class="logo__top-title"
            >
              THE
            </p>
            <h3
              class="logo__title"
            >
              FUTURE HUB
            </h3>
          </div>
        </a>
      </div>
      <ul
        class="nav-bar__link-list"
      >
        <a
          class="link"
          href="/"
        >
          <div
            class="nav-bar-item"
          >
            <div
              class="nav-bar-item__image"
            >
              <svg>
                dashboard-windows-square.svg
              </svg>
            </div>
            <p
              class="nav-bar-item__text"
            >
              Dashboard
            </p>
          </div>
        </a>
        <a
          class="link"
          href="/pathways"
        >
          <div
            class="nav-bar-item"
          >
            <div
              class="nav-bar-item__image"
            >
              <svg>
                pathways-chain-icon.svg
              </svg>
            </div>
            <p
              class="nav-bar-item__text"
            >
              Pathways
            </p>
          </div>
        </a>
        <a
          class="link"
          href="/articles"
        >
          <div
            class="nav-bar-item"
          >
            <div
              class="nav-bar-item__image"
            >
              <svg>
                article-book-icon.svg
              </svg>
            </div>
            <p
              class="nav-bar-item__text"
            >
              Articles
            </p>
          </div>
        </a>
        <a
          class="link"
          href="/marketplace"
        >
          <div
            class="nav-bar-item"
          >
            <div
              class="nav-bar-item__image"
            >
              <svg>
                marketplace-shopping-trolley-icon.svg
              </svg>
            </div>
            <p
              class="nav-bar-item__text"
            >
              Marketplace
            </p>
          </div>
        </a>
        <a
          class="link"
          href="/users"
        >
          <div
            class="nav-bar-item"
          >
            <div
              class="nav-bar-item__image"
            >
              <svg>
                users-people-icon.svg
              </svg>
            </div>
            <p
              class="nav-bar-item__text"
            >
              Users
            </p>
          </div>
        </a>
      </ul>
      <a
        class="link"
        href="/settings"
      >
        <div
          class="nav-bar-item"
        >
          <div
            class="nav-bar-item__image"
          >
            <svg>
              cog-icon.svg
            </svg>
          </div>
          <p
            class="nav-bar-item__text"
          >
            Settings
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;
